docs/README.md

```diff
@@ -0,0 +1,290 @@
+
+There's a lot more other pre-built subgraph queries that I could show you, but what I really want to show you is the `fetchCustomQuery` method.
+
+This allows you to fetch with your own queries and can be very powerful if you know graphQL. You can also check out a lot of example queries in our subgraph [playgrounds](https://thegraph.com/hosted-service/subgraph/toucanprotocol/matic).
+
+- Getting all infos on a project of a Carbon Credit
+
+```typescript
+import { gql } from "@urql/core";
+
+const query = gql`
+  query ($id: String) {
+    project(id: $id) {
+      projectId
+      region
+      standard
+      methodology
+      vintages {
+        id
+      }
+    }
+  }
+`;
+
+const result = await toucan.fetchCustomQuery(query, { id: "1" });
+```
+
+# AutoOffset using the OffsetHelper
+
+⚠️ This part is currently under construction and can't be used. ⚠️
+
+## AutoOffset CO2
+
+Using the OffsetHelper methods is the easiest way to offset CO2 right now as it handles the whole process for you.
+
+_(Uses Ubeswap/SushiSwap to acquire BCT/NCT, redeems it for TCO2s and retires them)_
+
+This is how you do that when you want to use ETH (or the network's native currency) in your balance.
+
+In this example, you'd be offsetting 1 TON of CO2 with carbon projects from the NCT pool.
+
+```typescript
+const offsetReceipt = await toucan.autoOffsetUsingETH("NCT", parseEther("1.0"));
+```
+
+### Don't want to spend your ETH to offset?
+
+You can use different tokens to do it. Currently it works with USDC, WMATIC and WETH only.
+
+```typescript
+const weth = new ethers.Contract(wethAddress, wethAbi, signer);
+
+const offsetReceipt = await toucan.autoOffsetUsingToken(
+  "NCT",
+  parseEther("1.0"),
+  weth
+);
+```
+
+### Offset with your own BCT/NCT
+
+If you already have BCT/NCT, you can use `autoOffsetPoolToken()` like so:
+
+```typescript
+const offsetReceipt = await toucan.autoOffsetPoolToken(
+  "NCT",
+  parseEther("1.0")
+);
+```
+
+### Selectively Offset CO2
+
+You may want to selectively offset certain carbon projects. You can do that too.
+
+Assuming you already have some NCT, this example gets an array of all TCO2s in the NCT pool ascending by rank/quality.
+
+Then redeems and retires 3 TONS of the highest ranked TCO2 it can find.
+
+```typescript
+const scoredTCO2s = await toucan.getScoredTCO2s("NCT");
+const len = scoredTCO2s.length;
+
+const redeemReceipt = await toucan.redeemMany(
+  "NCT",
+  [scoredTCO2s[len - 1]],
+  [parseEther("3.0")]
+);
+
+const retirementReceipt = await toucan.retire(
+  parseEther("3.0"),
+  tco2s[len - 1].address
+);
+```
+
+Of course, you could have avoided using the `getScoredTCO2s` method if you already had the address of the TCO2s you wanted to offset.
+
+### But I want the Retirement NFT
+
+If you do, that's also quite easy to pull off. You just replace the usage of the `retire()` method in the example above with this:
+
+```typescript
+const retirementReceipt = await toucan.retireAndMintCertificate(
+  "Alex",
```

Comment:
Let's use Alice here and below too. It's the standard way of naming stuff like this :)